docs/adr/assets/ADR-003/examples/nodejs/main.ts

```diff
@@ -1,79 +1,58 @@
-    login: string;
-  };
-}
+export const getOctokit = async (
+  appId: string,
+  privateKey: string,
+  orgName: string
+): Promise<RestOctokit> => {
+  const appOctokit = new Octokit({
+    authStrategy: createAppAuth,
+    auth: {
+      appId,
+      privateKey,
+    },
+  });
+  const installations = await appOctokit.request("GET /app/installations");
+  for (const d of installations.data) {
+    //@ts-ignore
+    if (d.account.login === orgName) {
+      const installationId = d.id;
+      const installationOctokit = new RestOctokit({
+        authStrategy: createAppAuth,
+        auth: {
+          appId,
+          privateKey,
+          installationId,
+        },
+      });
+      return installationOctokit;
+    }
+  }
+
+  throw new Error(`No installation found for organization ${orgName}`);
+};
 
 const ghAppId = process.env.GITHUB_APP_ID;
 const ghAppPkFile = process.env.GITHUB_APP_PK_FILE;
 const ghOrg = process.env.GITHUB_ORG;
 
-if (!ghAppId || !ghAppPkFile || !ghOrg) {
-  throw new Error(
-    "Environment variables GITHUB_APP_ID, GITHUB_APP_PK_FILE and GITHUB_ORG must be passed to this program."
-  );
-}
-
-function getJwtToken(ghAppId: string, ghAppPkFile: string): string {
-  const signingKey = fs.readFileSync(ghAppPkFile, "utf8");
-  const payload = {
-    iat: Math.floor(Date.now() / 1000),
-    exp: Math.floor(Date.now() / 1000) + 600,
-    iss: ghAppId,
-  };
-
-  return jwt.sign(payload, signingKey, { algorithm: "RS256" });
-}
-
-async function getInstallationId(
-  jwtToken: string,
-  ghOrg: string
-): Promise<number | null> {
-  const headers = {
-    Authorization: `Bearer ${jwtToken}`,
-    Accept: "application/vnd.github.v3+json",
-  };
-  const response = await axios.get<Installation[]>(
-    "https://api.github.com/app/installations",
-    { headers }
-  );
-  const installation = response.data.find(
-    (inst) => inst.account.login === ghOrg
-  );
-
-  return installation?.id || null;
-}
-
-async function getAccessToken(
-  jwtToken: string,
-  installationId: number
-): Promise<string> {
-  const headers = {
-    Authorization: `Bearer ${jwtToken}`,
-    Accept: "application/vnd.github.v3+json",
-  };
-  const response = await axios.post<{ token: string }>(
-    `https://api.github.com/app/installations/${installationId}/access_tokens`,
-    {},
-    { headers }
-  );
-
-  return response.data.token;
-}
-
 (async () => {
-  const jwtToken = getJwtToken(ghAppId, ghAppPkFile);
-  const installationId = await getInstallationId(jwtToken, ghOrg);
-  if (!installationId) {
-    console.log(`No installation found for organization ${ghOrg}`);
-    return;
+  if (!ghAppId || !ghAppPkFile || !ghOrg) {
+    throw new Error(
+      "Environment variables GITHUB_APP_ID, GITHUB_APP_PK_FILE, and GITHUB_ORG must be passed to this program."
+    );
   }
-  const accessToken = await getAccessToken(jwtToken, installationId);
+  const octokit = await getOctokit(
+    ghAppId,
+    fs.readFileSync(ghAppPkFile!, "utf8"),
```

Comment:
What does the exclamation mark do here?

Reply:
I had an "Argument of type 'string | undefined' is not assignable" compilation error here. Interesting, I've just removed the exclmation mark and it still works. Perhaps this is because I've done a bit of refactoring but now it is redundant. Thanks!